samcli/lib/providers/cfn_api_provider.py

```diff
@@ -75,6 +95,208 @@ def extract_resources(self, stacks: List[Stack], collector: ApiCollector, cwd: O
                 if resource_type == AWS_APIGATEWAY_V2_STAGE:
                     self._extract_cfn_gateway_v2_stage(resources, resource, collector)
 
+                if resource_type == AWS_APIGATEWAY_V2_AUTHORIZER:
+                    self._extract_cfn_gateway_v2_authorizer(logical_id, resource, collector)
+
+    @staticmethod
+    def _extract_cloud_formation_authorizer(logical_id: str, resource: dict, collector: ApiCollector) -> None:
+        """
+        Extract Authorizers from AWS::ApiGateway::Authorizer and add them to the collector.
+
+        Parameters
+        ----------
+        logical_id: str
+            The logical ID of the Authorizer
+        resource: dict
+            The attributes for the Authorizer
+        collector: ApiCollector
+            ApiCollector to save Authorizers into
+        """
+        properties = resource.get("Properties", {})
+        authorizer_type = properties.get(CfnApiProvider._AUTHORIZER_TYPE, "").lower()
+        rest_api_id = properties.get(CfnApiProvider._AUTHORIZER_REST_API)
+        name = properties.get(CfnApiProvider._AUTHORIZER_NAME)
+        authorizer_uri = properties.get(CfnApiProvider._AUTHORIZER_AUTHORIZER_URI)
+
+        if not authorizer_type:
+            raise InvalidSamTemplateException(
+                f"Authorizer '{logical_id}' is missing the '{CfnApiProvider._AUTHORIZER_TYPE}' "
+                "property, an Authorizer type must be defined."
+            )
+
+        if not rest_api_id:
+            raise InvalidSamTemplateException(
+                f"Authorizer '{logical_id}' is missing the '{CfnApiProvider._AUTHORIZER_REST_API}' "
+                "property, this must be defined."
+            )
+
+        if not name:
+            raise InvalidSamTemplateException(
+                f"Authorizer '{logical_id}' is missing the '{CfnApiProvider._AUTHORIZER_NAME}' "
+                "property, the Name must be defined."
+            )
+
+        if authorizer_type not in LambdaAuthorizer.VALID_TYPES:
+            LOG.warning(
+                "Authorizer '%s' with type '%s' is currently not supported. "
+                "Only Lambda Authorizers of type TOKEN and REQUEST are supported.",
+                logical_id,
+                authorizer_type,
+            )
+            return
+
+        if not authorizer_uri:
+            raise InvalidSamTemplateException(
+                f"Authorizer '{logical_id}' is missing the '{CfnApiProvider._AUTHORIZER_AUTHORIZER_URI}' "
+                "property, a valid Lambda ARN must be provided."
+            )
```

Comment:
I feel like this part can be extracted as validation method to to improve the readability. Which can also be combined what @sriram-mv suggested above (like different types might have their own implementation for a base interface or an abstract class).